converter/encoding_data_converter_test.go

```diff
@@ -205,30 +205,71 @@ func TestPayloadEncoderHTTPHandler(t *testing.T) {
 
 	require.Equal(t, http.StatusBadRequest, rr.Code)
 
-	payload, _ := defaultConv.ToPayload("test")
-	payloadJSON, _ := json.Marshal(payload)
+	payloads, _ := defaultConv.ToPayloads("test")
+	payloadsJSON, _ := json.Marshal(payloads)
 
-	fmt.Printf("%s", payloadJSON)
+	fmt.Printf("%s", payloadsJSON)
 
-	req, err = http.NewRequest("POST", "/encode", bytes.NewReader(payloadJSON))
+	req, err = http.NewRequest("POST", "/encode", bytes.NewReader(payloadsJSON))
 	if err != nil {
 		t.Fatal(err)
 	}
 	rr = httptest.NewRecorder()
 	handler.ServeHTTP(rr, req)
 
 	require.Equal(t, http.StatusOK, rr.Code)
-	encodedPayloadJSON := strings.TrimSpace(rr.Body.String())
-	require.NotEqual(t, payloadJSON, encodedPayloadJSON)
+	encodedPayloadsJSON := strings.TrimSpace(rr.Body.String())
+	require.NotEqual(t, payloadsJSON, encodedPayloadsJSON)
 
-	req, err = http.NewRequest("POST", "/decode", strings.NewReader(encodedPayloadJSON))
+	req, err = http.NewRequest("POST", "/decode", strings.NewReader(encodedPayloadsJSON))
 	if err != nil {
 		t.Fatal(err)
 	}
 	rr = httptest.NewRecorder()
 	handler.ServeHTTP(rr, req)
 
 	require.Equal(t, http.StatusOK, rr.Code)
-	decodedPayloadJSON := strings.TrimSpace(rr.Body.String())
-	require.Equal(t, string(payloadJSON), decodedPayloadJSON)
+	decodedPayloadsJSON := strings.TrimSpace(rr.Body.String())
+	require.Equal(t, string(payloadsJSON), decodedPayloadsJSON)
+}
+
+func TestRemoteDataConverter(t *testing.T) {
+	defaultConv := converter.GetDefaultDataConverter()
+	encoder := converter.NewZlibEncoder(converter.ZlibEncoderOptions{AlwaysEncode: true})
```

Comment:
Not necessarily needed _right now_, but ideally this test could have multiple encoders to test the loops are in the proper order.

Reply:
Adjusted the test to confirm this behaviour.